### PR TITLE
added docker - it's failed

### PR DESCRIPTION
**Pull request by yanna-stepanova**

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+version: '3.8'
+services:
+  mysqldb:
+    platform: linux/arm64
+    image: mysql:8.0.36
+    restart: unless-stopped
+    env_file: ./.env
+    environment:
+      MYSQL_ROOT_PASSWORD: '${MYSQLDB_PASSWORD}'
```
Comment (Elena-Bruyako) [inline docker-compose.yaml]:
```suggestion
      - MYSQL_ROOT_PASSWORD=$MYSQLDB_PASSWORD
```

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      MYSQL_DATABASE: '${MYSQLDB_DATABASE}'
+    ports:
+      - ${MYSQLDB_LOCAL_PORT}:${MYSQLDB_DOCKER_PORT}
+    expose:
```
Comment (Elena-Bruyako) [inline docker-compose.yaml]: remove

File: docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+version: '3.8'
+services:
+  mysqldb:
+    platform: linux/arm64
+    image: mysql:8.0.36
+    restart: unless-stopped
+    env_file: ./.env
+    environment:
+      MYSQL_ROOT_PASSWORD: '${MYSQLDB_PASSWORD}'
+      MYSQL_DATABASE: '${MYSQLDB_DATABASE}'
```
Comment (Elena-Bruyako) [inline docker-compose.yaml]:
```suggestion
      - MYSQL_DATABASE=$MYSQLDB_DATABASE
```

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      retries: 3
+  app:
+    depends_on:
+      - mysqldb
+    restart: on-failure
+    image: yanna-book-store:1.0
+    build: .
+    env_file: ./.env
+    ports:
+      - ${SPRING_LOCAL_PORT}:${SPRING_DOCKER_PORT}
```
Comment (Elena-Bruyako) [inline docker-compose.yaml]:
```suggestion
      - $SPRING_LOCAL_PORT:$SPRING_DOCKER_PORT
```

File: docker-compose.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      - ${DEBUG_PORT}:${DEBUG_PORT}
```
Comment (Elena-Bruyako) [inline docker-compose.yaml]:
```suggestion
      - $DEBUG_PORT:$DEBUG_PORT
```

File: docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    environment:
+      SPRING_APPLICATION_JSON: '{
+        "spring.datasource.url"  : "jdbc:mysql://mysqldb:${MYSQLDB_DOCKER_PORT}/${MYSQLDB_DATABASE}",
```
Comment (Elena-Bruyako) [inline docker-compose.yaml]:
```suggestion
        "spring.datasource.url"  : "jdbc:mysql://mysqldb:$MYSQLDB_DOCKER_PORT/$MYSQLDB_DATABASE",
```

File: docker-compose.yaml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        "spring.datasource.username" : "${MYSQLDB_USER}",
```
Comment (Elena-Bruyako) [inline docker-compose.yaml]:
```suggestion
        "spring.datasource.username" : "$MYSQLDB_USER",
```

File: docker-compose.yaml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        "spring.datasource.password" : "${MYSQLDB_PASSWORD}",
```
Comment (Elena-Bruyako) [inline docker-compose.yaml]:
```suggestion
        "spring.datasource.password" : "$MYSQLDB_PASSWORD",
```

File: pom.xml [hunk 9]
```diff
@@ -132,7 +132,10 @@
 			<version>${jjwt.version}</version>
 			<scope>runtime</scope>
 		</dependency>
-
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-docker-compose</artifactId>
```
Comment (Elena-Bruyako) [inline pom.xml]: remove

File: Dockerfile [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+# Final stage
+FROM openjdk:17-jdk-slim
+WORKDIR application
+COPY --from=builder application/dependencies/ ./
+COPY --from=builder application/spring-boot-loader/ ./
+COPY --from=builder application/snapshot-dependencies/ ./
+COPY --from=builder application/application/ ./
+ENTRYPOINT ["java", "org.springframework.boot.loader.launch.JarLauncher"]
+EXPOSE 8080
+EXPOSE 5005
```
Comment (Elena-Bruyako) [inline Dockerfile]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Left some comments

File: docker-compose.yaml (hunk 2 above)
Comment (yanna-stepanova) [reply docker-compose.yaml]: is it about line 13+14?

File: docker-compose.yaml [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+      - $DEBUG_PORT:$DEBUG_PORT
+    environment:
+      SPRING_APPLICATION_JSON: '{
+        "spring.datasource.url"  : "jdbc:mysql://mysqldb:$MYSQLDB_DOCKER_PORT/$MYSQLDB_DATABASE",
+        "spring.datasource.username" : "$MYSQLDB_USER",
+        "spring.datasource.password" : "$MYSQLDB_PASSWORD",
+        "spring.jpa.database-platform" : "org.hibernate.dialect.MySQLDialect",
+        "spring.datasource.driver-class-name" : "com.mysql.cj.jdbc.Driver"
+      }'
+      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005"
```
Comment (okuzan) [inline docker-compose.yaml]: 5005 - it's a debug port, use .env var

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by okuzan — APPROVED** [review]

**Pull request merged**